nilearn/_utils/param_validation.py

```diff
@@ -60,28 +65,50 @@ def check_threshold(threshold, data, percentile_func, name="threshold"):
             raise ValueError(message)
 
         try:
-            percentile = float(threshold[:-1])
+            threshold = float(threshold[:-1])
+            percentile = True
         except ValueError as exc:
             exc.args += (message,)
             raise
-
-        threshold = percentile_func(data, percentile)
-    elif isinstance(threshold, numbers.Real):
-        # checks whether given float value exceeds the maximum
-        # value of the image data
-        value_check = abs(data).max()
-        if abs(threshold) > value_check:
-            warnings.warn(
-                f"The given float value must not exceed {value_check}. "
-                f"But, you have given threshold={threshold}.",
-                category=UserWarning,
-                stacklevel=3,
-            )
-    else:
+    elif not isinstance(threshold, numbers.Real):
         raise TypeError(
             f"{name} should be either a number "
             "or a string finishing with a percent sign"
         )
+
+    if threshold >= 0:
+        data = abs(data) if two_sided else np.extract(data >= 0, data)
+
+        if percentile:
+            threshold = percentile_func(data, threshold)
+        else:
+            value_check = data.max()
+            if threshold > value_check:
+                warnings.warn(
+                    f"The given float value must not exceed {value_check}. "
+                    f"But, you have given threshold={threshold}.",
+                    category=UserWarning,
+                    stacklevel=3,
+                )
+    else:
+        if two_sided:
+            raise ValueError(
+                f'"{name}" should not be a negative value when two_sided=True.'
+            )
+        data = np.extract(data <= 0, data)
+        if percentile:
+            threshold = -1 * percentile_func(abs(data), abs(threshold))
```

Comment:
I'm unsure we really want to do this. I think that we should raise an error if a negative percentile if provided.

Reply:
I was not sure if it could be interesting to get only negative regions smaller than some certain threshold. 

If it is not relevant at all, I can remove it.

Reply:
Is it only negative percentile, or negative number as well?

Reply:
Negative percentile for sure.
Negative value, I would keep it, because we keep the standard behavior.

Reply:
OK, I remove negative percentile then.

Reply:
Just to make sure that the behavior of negative threshold is as expected:

It is only possible for `two_sided=False`.

The values above negative threshold are removed.